queries.sql

```diff
@@ -5,4 +5,22 @@ SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name = 'Pikachu'; --
 SELECT name, escape_attempts FROM animals WHERE weight_in_kg > 10.5; -- List name and escape attempts of animals that weigh more than 10.5kg
 SELECT * FROM animals WHERE neutered = TRUE; -- Find all animals that are neutered.
 SELECT * FROM animals WHERE NOT name = 'Gabumon'; -- Find all animals not named Gabumon.
-SELECT * FROM animals WHERE weight_in_kg >= 10.4 AND weight_in_kg <= 17.3; -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
\ No newline at end of file
+SELECT * FROM animals WHERE weight_in_kg >= 10.4 AND weight_in_kg <= 17.3; -- Find all animals with a weight between 10.4kg and 17.3kg (including the animals with the weights that equals precisely 10.4kg or 17.3kg)
+
+SELECT COUNT(*) AS totalAnimals FROM animals; -- How many animals are there?
+SELECT COUNT(*) FROM animals WHERE escape_attempts > 0; -- How many animals have never tried to escape?
+SELECT AVG(weight_in_kg) FROM animals; -- What is the average weight of animals?
+
+SELECT
+CASE WHEN neutered = TRUE THEN 'neutered' ELSE 'not neutered' END AS is_neutered
+FROM 
+( 
+  SELECT neutered , SUM(escape_attempts) as total
+  FROM animals 
+  GROUP BY neutered 
+  ORDER BY total DESC
+) AS animals
+LIMIT 1; -- Who escapes the most, neutered or not neutered animals?
+
+SELECT species, Min(weight_in_kg) AS min_weight, Max(weight_in_kg) AS max_weight FROM animals GROUP BY species; -- What is the minimum and maximum weight of each type of animal?
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth >= '1990-01-01' AND date_of_birth <= '2000-01-01' GROUP BY species;-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
```

Comment:
- [ ] To find the average number of escape attempts per animal type of those born between 1990 and 2000, the first day of 1990 and the last day of 2000 should also be included in the query e.g.
```sql
SELECT species, AVG(escape_attempts) FROM animals GROUP BY species
   WHERE  date_of_birth >= '1990/01/01' AND date_of_birth <= '2000/31/12';
```